File: shared/src/commonMain/kotlin/nl/fbdevelopment/healthyplatform/api/service/AuthService.kt
package nl.fbdevelopment.healthyplatform.api.service

import io.ktor.client.statement.HttpResponse
import nl.fbdevelopment.healthyplatform.data.models.dto.auth.AuthTokenDto
import nl.fbdevelopment.healthyplatform.data.models.dto.auth.LoginDto
import nl.fbdevelopment.healthyplatform.data.models.dto.auth.RegistrationDto

interface AuthService {
    suspend fun login(body: LoginDto) : HttpResponse
    suspend fun logout() : HttpResponse
    suspend fun register(body: RegistrationDto): HttpResponse
    suspend fun refreshTokens() : HttpResponse
    suspend fun session(): HttpResponse
}

File: webApp/src/api/service/AuthService.ts
// AuthService.ts
import api from "../axios";
import type { UserWithContext } from "../../data/types";

export const AuthService = {
  login(email: string, password: string) {
    return api.post<UserWithContext>("/auth/login", { email, password }).then((r) => r.data);
  },

  register(username: string, email: string, password: string) {
    return api.post<UserWithContext>("/auth/register", { username, email, password }).then((r) => r.data);
  },

  fetchSession() {
    return api.get<UserWithContext>("/auth/session").then((r) => r.data);
  },

  refreshSession() {
    return api.post("/auth/refresh").then(() => {});
  },

  logout() {
    return api.post("/auth/logout").then(() => {});
  },
};
